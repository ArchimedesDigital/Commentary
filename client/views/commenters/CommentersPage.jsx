CommentersPage = React.createClass({

	propTypes: {},

	render() {
		return (
			<div className="page page-commenters">
				<div className="content primary">
					<section className="block header cover parallax">
						<div className="background-image-holder blur-2--no-remove blur-10 remove-blur">
							<img
								className="background-image"
								src="/images/capitals.jpg"
								alt="Commenter Page | A Homer Commentary in Progress"
							/>
						</div>
						<div className="block-screen brown"/>
						<div className="container v-align-transform">
							<div className="grid inner">
								<div className="center-content">
									<div className="page-title-wrap">
										<h2 className="page-title ">
											Commenters
										</h2>
									</div>
								</div>
							</div>
						</div>
					</section>
					<section className="page-content">

						<CommentersList defaultAvatarUrl="/images/default_user.jpg"/>

					</section>

					<CommentsRecent />
				</div>

			</div>
		);
	},
});
