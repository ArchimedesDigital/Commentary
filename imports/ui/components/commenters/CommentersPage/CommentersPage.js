import React, { Component } from 'react';
import { Meteor } from 'meteor/meteor';
import { Session } from 'meteor/session';
import { createContainer } from 'meteor/react-meteor-data';

// components
import BackgroundImageHolder from '/imports/ui/components/shared/BackgroundImageHolder';
import CommentersList from '/imports/ui/components/commenters/CommentersList';
import CommentsRecent from '/imports/ui/components/commentary/comments/CommentsRecent';

// models
import Settings from '/imports/models/settings';

// lib
import Utils from '/imports/lib/utils';

class CommentersPage extends Component {

	render() {
		const { settings } = this.props;

		if (!settings) {
			return null;
		}

		Utils.setTitle(`Commentators | ${settings.title}`);
		Utils.setDescription(`Commentators for ${settings.title}`);
		Utils.setMetaImage(`${location.origin}/images/capitals.jpg`);
		return (
			<div className="page page-commenters">
				<div className="content primary">
					<section className="block header cover parallax">
						<BackgroundImageHolder
							imgSrc="/images/capitals.jpg"
						/>
						<div className="container v-align-transform">
							<div className="grid inner">
								<div className="center-content">
									<div className="page-title-wrap">
										<h2 className="page-title ">
											Commentators
										</h2>
									</div>
								</div>
							</div>
						</div>
					</section>
					<section className="page-content">

						<CommentersList />

					</section>

					<CommentsRecent />
				</div>

			</div>
		);
	}
}

CommentersPage.propTypes = {
	settings: React.PropTypes.object
};

const commentersPageContainer = createContainer(() => {
	const settingsHandle = Meteor.subscribe('settings.tenant', Session.get('tenantId'));

	return {
		settings: settingsHandle.ready() ? Settings.findOne() : { title: '' }
	};
}, CommentersPage);


export default commentersPageContainer;
