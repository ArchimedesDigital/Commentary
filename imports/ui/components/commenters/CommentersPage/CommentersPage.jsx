import React from 'react';
import { Meteor } from 'meteor/meteor';
import { Session } from 'meteor/session';
import { createContainer } from 'meteor/react-meteor-data';
import BackgroundImageHolder from '/imports/ui/components/shared/BackgroundImageHolder';

const CommentersPage = React.createClass({

	render() {
		const { settings } = this.data;
		Utils.setTitle(`Commentators | ${settings.title}`);
		Utils.setDescription(`Commentators for ${Config.title}`);
		Utils.setMetaImage(`${location.origin}/images/capitals.jpg`);
		return (
			<div className="page page-commenters">
				<div className="content primary">
					<section className="block header cover parallax">
						<BackgroundImageHolder
							imgSrc="/images/capitals.jpg"
						/>
						<div className="container v-align-transform">
							<div className="grid inner">
								<div className="center-content">
									<div className="page-title-wrap">
										<h2 className="page-title ">
											Commentators
										</h2>
									</div>
								</div>
							</div>
						</div>
					</section>
					<section className="page-content">

						<CommentersList defaultAvatarUrl="/images/default_user.jpg" />

					</section>

					<CommentsRecent />
				</div>

			</div>
		);
	},
});

const commentersPageContainer = createContainer(() => {
	const settingsHandle = Meteor.subscribe('settings.tenant', Session.get('tenantId'));

	return {
		settings: settingsHandle.ready() ? Settings.findOne() : { title: '' }
	};
}, CommentersPage);


export default commentersPageContainer;
