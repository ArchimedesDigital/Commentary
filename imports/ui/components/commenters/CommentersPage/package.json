{
  "name": "CommentersPage",
  "version": "0.0.1",
  "private": true,
  "main": "./CommentersPage.js"
}
