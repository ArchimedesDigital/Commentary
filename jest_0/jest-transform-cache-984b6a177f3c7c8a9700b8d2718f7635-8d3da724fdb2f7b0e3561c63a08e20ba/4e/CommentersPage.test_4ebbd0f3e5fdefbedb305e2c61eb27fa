/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommentersPage = require('./CommentersPage'); /* istanbul ignore next */var _CommentersPage2 = _interopRequireDefault(_CommentersPage);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommentersPage', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommentersPage2.default, /* istanbul ignore next */null)).
		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: