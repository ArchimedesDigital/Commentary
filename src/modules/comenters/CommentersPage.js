import React, { Component } from 'react';
import PropTypes from 'prop-types';

import muiTheme from '../../lib/muiTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { compose } from 'react-apollo';

// components
import BackgroundImageHolder from '../shared/BackgroundImageHolder';
import CommentersList from '../comenters/CommentersList';
import CommentsRecent from '../comments/commentsRecent/CommentsRecent';
import Header from '/imports/ui/layouts/header/Header';

// graphql
import { settingsQuery } from '../../graphql/methods/settings';

// lib
import Utils from '../../lib/utils';

class CommentersPage extends Component {

	constructor(props) {
		super(props);
		this.state = {};
	}
	componentWillReceiveProps(nextProps) {
		const tenantId = sessionStorage.getItem('tenantId');
		const settings = nextProps.settingsQuery.loading ? { title: ''} : nextProps.settingsQuery.settings.find(x => x.tenantId === tenantId);
		this.setState({
			settings: settings
		});
	}
	render() {
		const { settings } = this.state;

		if (!settings) {
			return null;
		}

		Utils.setTitle(`Commentators | ${settings.title}`);
		Utils.setDescription(`Commentators for ${settings.title}`);
		Utils.setMetaImage(`${window.location.origin}/images/capitals.jpg`);
		return (
			<MuiThemeProvider>
				<div className="page page-commenters">
					<Header />
					<div className="content primary">
						<section className="block header cover parallax">
							<BackgroundImageHolder
								imgSrc="/images/capitals.jpg"
							/>
							<div className="container v-align-transform">
								<div className="grid inner">
									<div className="center-content">
										<div className="page-title-wrap">
											<h2 className="page-title ">
												Commentators
											</h2>
										</div>
									</div>
								</div>
							</div>
						</section>
						<section className="page-content">

							<CommentersList />

						</section>

						<CommentsRecent />
					</div>

				</div>
			</MuiThemeProvider>
		);
	}
}

CommentersPage.propTypes = {
	settingsQuery: PropTypes.object
};


export default compose(settingsQuery)(CommentersPage);
