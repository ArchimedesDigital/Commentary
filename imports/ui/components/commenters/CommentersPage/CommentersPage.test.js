import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentersPage from './CommentersPage';

describe('CommentersPage', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentersPage />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
