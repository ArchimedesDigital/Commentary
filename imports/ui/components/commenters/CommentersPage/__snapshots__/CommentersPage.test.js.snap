// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentersPage renders correctly 1`] = `null`;
